src/flow-util.c

```diff
@@ -153,6 +153,9 @@ void FlowInit(ThreadVars *tv, Flow *f, const Packet *p)
     f->recursion_level = p->recursion_level;
     memcpy(&f->vlan_id[0], &p->vlan_id[0], sizeof(f->vlan_id));
     f->vlan_idx = p->vlan_idx;
+
+    f->thread_id[0] = (FlowThreadId)tv->id;
```

Comment:
Curious why this is hard coded for toserver instead of the pkt direction?

Reply:
We don't know the packet direction when we create the flow, we assume the first packet is to server. We may later reverse it based on a few checks (tcp syn/ack, proto detect).